src/scaler/utility/queues/async_priority_queue.py

```diff
@@ -1,70 +1,83 @@
-import heapq
-import sys
 from asyncio import Queue
-from typing import Any, Dict, List, Tuple, Union
+from dataclasses import dataclass
+from typing import Any, Dict, Tuple, Union
+
+from sortedcontainers import SortedDict
 
 PriorityType = Union[int, Tuple["PriorityType", ...]]
 
 
 class AsyncPriorityQueue(Queue):
     """A subclass of Queue; retrieves entries in priority order (lowest first).
 
-    Entries are typically list of the form: [priority, data].
+    Input entries are typically list of the form: [priority, data].
     """
 
+    @dataclass(frozen=True)
+    class MapKey:
+        priority: int
+        count: int
+
+        def __lt__(self, other):
+            return (self.priority, self.count) < (other.priority, other.count)
+
+        def __hash__(self):
+            return hash((self.priority, self.count))
+
+    @dataclass
+    class LocatorValue:
+        map_key: "AsyncPriorityQueue.MapKey"
```

Comment:
```suggestion
        map_key: AsyncPriorityQueue.MapKey
```

Doesn't that work too?

Reply:
No.
```
line 29, in LocatorValue
    map_key: AsyncPriorityQueue.MapKey
             ^^^^^^^^^^^^^^^^^^
NameError: name 'AsyncPriorityQueue' is not defined
```